File: Programmers/모의고사.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;
int su1[5] = { 1,2,3,4,5 };
int su2[8] = { 2,1,2,3,2,4,2,5 };
int su3[10] = { 3,3,1,1,2,2,4,4,5,5 };

vector<int> solution(vector<int> answers) {
	vector<int> answer;
	vector<int> cnt(3);
	for (int i = 0; i < answers.size(); i++) {
		if (answers[i] == su1[i % 5]) cnt[0]++;
		if (answers[i] == su2[i % 8]) cnt[1]++;
		if (answers[i] == su3[i % 10]) cnt[2]++;
	}
	int MAX = -1;
	MAX = max(max(cnt[0], cnt[1]), cnt[2]);
	for (int i = 0; i < 3; i++) {
		if (MAX == cnt[i]) answer.push_back(i + 1);
	}
	return answer;
}

File: Programmers/python/모의고사.py
def solution(answers):
    supo1 = [1, 2, 3, 4, 5]
    supo2 = [2, 1, 2, 3, 2, 4, 2, 5]
    supo3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    score = [0, 0, 0]
    
    for i in range(len(answers)):
        if answers[i] == supo1[i%5]:
            score[0] += 1
        if answers[i] == supo2[i%8]:
            score[1] += 1
        if answers[i] == supo3[i%10]:
            score[2] += 1
            
    answer = []
    for i in range(3):
        if score[i] == max(score):
            answer.append(i+1)
            
    return answer